### ISSUE TITLE
Qwen3.5 Performance Report

### ISSUE THREAD

**Issue by YichiZhang02** [starVLA/starVLA#291]
I conducted experiments using StarVLA with Qwen3.5-0.8B, and the results are shown below.

<img width="1101" height="197" alt="Image" src="https://github.com/user-attachments/assets/b03cccfe-fcae-4014-b902-b0727b6580d6" />

Additionally, I would like to report a bug: the special tokens used in QwenOFT are decoded into three separate tokens by the Qwen3.5 tokenizer. This leads to issues in processing action queries. After fixing this problem, we observe a significant improvement in performance.

Here is my fix in QwenOFT:
```python
self.action_token = "<|fim_pad|>"
self.action_token_id = self.qwen_vl_interface.processor.tokenizer(
    "<|fim_pad|>", add_special_tokens=False
)["input_ids"][0]
```

**Comment by JinhuiYE** [comment]
Thank you for sharing your experimental results with StarVLA using Qwen3.5-0.8B.

We welcome you to submit a PR under `examples/LIBERO/README.md` to report your performance and upload your results.

We also recommend that you upload your checkpoint to your own Hugging Face (HF) and add a hyperlink to `examples/LIBERO/README.md`.

Regarding `action_token`: since our current model has fixed tokens, you may need to add a patch to use your tokens when handling the Qwen3.5 VLM.

**Comment by haofuly** [comment]
Hi @ChiGod0502 , thanks for your contribution. If I want to use Qwen3.5 instead of Qwen3, I only need to change the `base_vlm checkpoint path` and install the `flash-linear-attention`. Is it right? Thanks a lot.

**Comment by YichiZhang02** [comment]
> Hi [@ChiGod0502](https://github.com/ChiGod0502) , thanks for your contribution. If I want to use Qwen3.5 instead of Qwen3, I only need to change the `base_vlm checkpoint path` and install the `flash-linear-attention`. Is it right? Thanks a lot.

Yes, by doing so, you can train and test correctly, but you may encounter a precessor warning. You can adjust the input kwargs of the qwen3.5 processor to solve this problem.

```python
**Kwargs passed to `processor.__call__` have to be in `processor_kwargs` dict, not in `kwargs`**
```

**Comment by haofuly** [comment]
I will try it. Thanks!

**Comment by HanfengLiao** [comment]
> I conducted experiments using StarVLA with Qwen3.5-0.8B, and the results are shown below.
> 
> <img alt="Image" width="1101" height="197" src="https://private-user-images.githubusercontent.com/175929104/583860089-b03cccfe-fcae-4014-b902-b0727b6580d6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODA2NDE0MzIsIm5iZiI6MTc4MDY0MTEzMiwicGF0aCI6Ii8xNzU5MjkxMDQvNTgzODYwMDg5LWIwM2NjY2ZlLWZjYWUtNDAxNC1iOTAyLWIwNzI3YjY1ODBkNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDYwNVQwNjMyMTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wY2I4MDFmNTM0NWQ5ODI4YTNhZjkwOWYzOGFlOTJjM2M1MDJlYmI0NWVlOWExZTA3MmVjODdjYzljNzA4ZGM2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.2nuvOvNrpbIJiE_8Vg9URIzV9f7sP-XIKu1Alvk42fo">
> Additionally, I would like to report a bug: the special tokens used in QwenOFT are decoded into three separate tokens by the Qwen3.5 tokenizer. This leads to issues in processing action queries. After fixing this problem, we observe a significant improvement in performance.
> 
> Here is my fix in QwenOFT:
> 
> self.action_token = "<|fim_pad|>"
> self.action_token_id = self.qwen_vl_interface.processor.tokenizer(
>     "<|fim_pad|>", add_special_tokens=False
> )["input_ids"][0]

Hi @ChiGod0502 , I trained oft and gr00t with qwen3.5 0.8B follow your set up(total bs=16 and total steps=80k) and fixing. And I got this result which is far lower than your report:
gadi h200 | spatial | object | goal | long | Avg
-- | -- | -- | -- | -- | --
oft baseline | 78.80% | 81.80% | 74.60% | 44.60% | 69.95%
gr00t,baseline | 71.4% | 79% | 70.8% | 45% | 66.55

Could you comfirm that the "bs total" is really overall total or per device bs total? It is confusing because the default set up for starvla is 8 GPUs with 16 bs per GPU. 

By the way, the final action loss is around 0.07 from my training. Can I know the final action loss of your training?

Thx

**Comment by YichiZhang02** [comment]
> > I conducted experiments using StarVLA with Qwen3.5-0.8B, and the results are shown below.
> > <img alt="Image" width="1101" height="197" src="https://private-user-images.githubusercontent.com/175929104/583860089-b03cccfe-fcae-4014-b902-b0727b6580d6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODA2NDE0MzIsIm5iZiI6MTc4MDY0MTEzMiwicGF0aCI6Ii8xNzU5MjkxMDQvNTgzODYwMDg5LWIwM2NjY2ZlLWZjYWUtNDAxNC1iOTAyLWIwNzI3YjY1ODBkNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDYwNVQwNjMyMTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wY2I4MDFmNTM0NWQ5ODI4YTNhZjkwOWYzOGFlOTJjM2M1MDJlYmI0NWVlOWExZTA3MmVjODdjYzljNzA4ZGM2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.2nuvOvNrpbIJiE_8Vg9URIzV9f7sP-XIKu1Alvk42fo">
> > Additionally, I would like to report a bug: the special tokens used in QwenOFT are decoded into three separate tokens by the Qwen3.5 tokenizer. This leads to issues in processing action queries. After fixing this problem, we observe a significant improvement in performance.
> > Here is my fix in QwenOFT:
> > self.action_token = "<|fim_pad|>"
> > self.action_token_id = self.qwen_vl_interface.processor.tokenizer(
> > "<|fim_pad|>", add_special_tokens=False
> > )["input_ids"][0]
> 
> Hi [@ChiGod0502](https://github.com/ChiGod0502) , I trained oft and gr00t with qwen3.5 0.8B follow your set up(total bs=16 and total steps=80k) and fixing. And I got this result which is far lower than your report:
> 
> gadi h200	spatial	object	goal	long	Avg
> oft baseline	78.80%	81.80%	74.60%	44.60%	69.95%
> gr00t,baseline	71.4%	79%	70.8%	45%	66.55
> Could you comfirm that the "bs total" is really overall total or per device bs total? It is confusing because the default set up for starvla is 8 GPUs with 16 bs per GPU.
> 
> By the way, the final action loss is around 0.07 from my training. Can I know the final action loss of your training?
> 
> Thx

Hi, thanks for trying to reproduce the results.

Yes, the reported total batch size is 16 overall, not 16 per GPU. In our experiments, we only used 2× RTX 5090 GPUs, so the effective global batch size was 16. That said, this setup may indeed make the results somewhat harder to reproduce due to the relatively small batch size. You may want to increase the total batch size if resources allow, as we found that larger batch sizes generally lead to more stable training.

Regarding the action loss, our final action loss is typically around 0.01–0.02 (Gr00t). Since your training ends at approximately 0.07, my guess is that the model may not have fully converged yet, which could partly explain the performance gap.

<img width="2505" height="1083" alt="Image" src="https://github.com/user-attachments/assets/d02a1c6a-ea64-4a25-b8cf-50718d345622" />

**Comment by fyx-doe** [comment]
Hello, I’m also trying to reproduce your results with the GR00T action head, but my trained model still gets a 0 success rate during evaluation. May I ask what specific training configuration you used?And my loss also does not reach your level. May I also ask if your training was done using the libero_all dataset?

**Comment by YichiZhang02** [comment]
> Hello, I’m also trying to reproduce your results with the GR00T action head, but my trained model still gets a 0 success rate during evaluation. May I ask what specific training configuration you used?And my loss also does not reach your level. May I also ask if your training was done using the libero_all dataset?

Hi,

For the GR00T action head, I used the default training configuration provided in the repository and did not make any special modifications to the hyperparameters. The training was conducted on the libero_all dataset.

You may also want to double-check that your environment setup matches the repository requirements. As a sanity check, I would recommend downloading the released pretrained weights and running evaluation first. If the pretrained model also gets a 0 success rate, the issue is likely related to the environment or evaluation setup rather than training.

**Comment by HanfengLiao** [comment]
> Hello, I’m also trying to reproduce your results with the GR00T action head, but my trained model still gets a 0 success rate during evaluation. May I ask what specific training configuration you used?And my loss also does not reach your level. May I also ask if your training was done using the libero_all dataset?

Hi there, I saw 0 SR occurring to you which was the problem I had. Pls check the output eval video and see the general trajectory. The most possible point is about dataset, the training action label especially the gripper threshold range is different from the eval env.
